File: src/web/src/Shared/Models/Schedule.ts
export const LectureCell: string = "LECTURE"
export const PracticeCell: string = "PRACTICE"
export const SeminarCell: string = "SEMINAR"
export const laboratoryCell: string = "LABORATORY"

/**
 * Schedule day representation
 */
export interface ScheduleDay {
    date: number
    lessons: Lesson[]
}

/**
 * Lesson model
 */
export interface Lesson {
    id: string
    title: string
    position: number
    lessonType: string
    audience: Audience
    groups: GroupInfo[]
    professor: Professor
}

/**
 * Audience info
 */
export interface Audience {
    id: string
    name: string
}

/**
 * Group Info
 */
export interface GroupInfo {
    id: string
    name: string
}


/**
 * Faculty Info
 */
export interface FacultyInfo {
    id: string
    name: string
}

/**
 * Professor info
 */
export interface Professor {
    id: string
    name: string
}

File: ExtendedSchedule/domain/entity/schedule.go
package entity

import "time"

const (
	EmptyLesson    = "EMPTY"
	NotEmptyLesson = "LESSON"
)

// AudienceInfo defines model for audience info.
type AudienceInfo struct {
	ID   string
	Name string
}

// GroupInfo defines model for group info.
type GroupInfo struct {
	ID   string
	Name string
}

// Lesson defines model for lesson.
type Lesson struct {
	ID           string
	Title        string
	LessonNumber int
	LessonType   string
	Audience     AudienceInfo
	Groups       []GroupInfo
}

// DaySchedule defines model for day schedule.
type DaySchedule struct {
	Date     time.Time
	Sections []Section
}

type Section struct {
	Position int
	Lessons  []Lesson
}
